### PR TITLE
match? in an `is` returns boolean

### PR DESCRIPTION
**Pull request by dchelimsky**
Solution for https://github.com/nubank/matcher-combinators/issues/109

### CODE REVIEW

File: test/clj/matcher_combinators/test_test.clj [hunk 1]
```diff
@@ -17,12 +17,22 @@
...
+
 (deftest basic-matching
   (is (match? example-matcher example-actual)
       "In 'match?', the matcher argument comes first")
   (is (match? (m/equals example-matcher)
               (dissoc example-actual :device))
       "wrapping the matcher in 'equals' means the top level of 'actual' must have the exact same key/values")
+  (is (true? (is (match? 1 1)))
+      "match? should return true for a :match")
+  (is (false? (shhh! (is (match? 1 2))))
```
Comment (markomafs) [inline test/clj/matcher_combinators/test_test.clj]: Now, that means we can assert like `(not (match? ... )) ` 

How about add a test for this? 🤔🤔🤔

Comment (dchelimsky) [reply test/clj/matcher_combinators/test_test.clj]: That is the test! Unless you want me to add a test that Clojure works as documented w/ regard to `not` `nil` and `false` ;)

Comment (dchelimsky) [reply test/clj/matcher_combinators/test_test.clj]: We _could_ add it to the docs, though.

Comment (markomafs) [reply test/clj/matcher_combinators/test_test.clj]: Works for me, mainly because we didn’t support that in the past.

Comment (dchelimsky) [reply test/clj/matcher_combinators/test_test.clj]: Actually, that won't work. `match?` is not a function - it's an `assert-expr` in clojure.test, which only works if it's the first symbol in the seq passed to `is` e.g. `(is (match? ..))`. If we were to say `(is (not (match? ...)))` we'd get an error on `match?` being unresolved.

We could add special handling for `(is (not (match? ...)))` but that would require another PR.
